plugins/com.gwtplugins.gwt.eclipse.core/src/com/google/gwt/eclipse/core/validators/java/JavaCompilationParticipant.java

```diff
@@ -412,7 +411,7 @@ public void acceptAST(ICompilationUnit source, CompilationUnit ast) {
               // Record the JSNI dependencies so that changing any referenced
               // types
               // will automatically trigger a rebuild of this file
-              buildContext.recordDependencies(typeDependencies.toArray(Empty.STRINGS));
+              buildContext.recordDependencies(typeDependencies.toArray(new String[typeDependencies.size()]));
```

Comment:
Instead, this can be an empty String[], as the array won't be used except to check the type. Reusing an existing one (like Empty.STRINGS previously did) just saves even a tiny extra allocation instead of a potentially bigger one.

Reply:
Good catch,
Using Empty.STRINGS will save one allocation in case of empty typeDependencies,

Reply:
Note that we no longer need this.